azurelinuxagent/ga/collect_logs.py

```diff
@@ -128,7 +130,10 @@ def stopped(self):
     def stop(self):
         self.should_run = False
         if self.is_alive():
-            self.join()
+            try:
+                self.join()
+            except RuntimeError:
```

Comment:
RuntimeError being raised when agent try to stop the thread to exit gracefully. Seems like expected error ignoring this

  File "bin/WALinuxAgent-9.9.9.9-py3.8.egg/azurelinuxagent/ga/collect_logs.py", line 123, in join
    self.event_thread.join()
  File "/usr/lib/python3.6/threading.py", line 1053, in join
    raise RuntimeError("cannot join current thread")
RuntimeError: cannot join current thread